model/lifecycle.go

```diff
@@ -1314,6 +1296,58 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 	return t, nil
 }
 
+func generateDistros(id string, buildVarTask BuildVariantTaskUnit, buildVariant *BuildVariant, project *Project, v *Version) (string, []string) {
+	var (
+		distroID      string
+		distroAliases []string
+	)
+
+	if len(buildVarTask.RunOn) > 0 {
+		distroID = buildVarTask.RunOn[0]
+
+		if len(buildVarTask.RunOn) > 1 {
+			distroAliases = append(distroAliases, buildVarTask.RunOn[1:]...)
+		}
+
+	} else if len(buildVariant.RunOn) > 0 {
+		distroID = buildVariant.RunOn[0]
+
+		if len(buildVariant.RunOn) > 1 {
+			distroAliases = append(distroAliases, buildVariant.RunOn[1:]...)
+		}
+	} else {
+		grip.Warning(message.Fields{
+			"task_id":   id,
+			"message":   "task is not runnable as there is no distro specified",
+			"variant":   buildVariant.Name,
+			"project":   project.Identifier,
+			"version":   v.Revision,
+			"requester": v.Requester,
+		})
+	}
+	return distroID, distroAliases
+}
+
+func generateContainer(id string, buildVarTask BuildVariantTaskUnit, buildVariant *BuildVariant, project *Project, v *Version) string {
+	var container string
+
+	if len(buildVarTask.RunOn) > 0 {
+		container = buildVarTask.RunOn[0]
+	} else if len(buildVariant.RunOn) > 0 {
+		container = buildVariant.RunOn[0]
```

Comment:
It seems like this will be weird unless users are disallowed from passing multiple containers or a mix of containers/distros to `RunOn`. Is the project validation for RunOn going to be changed to make sure this is guaranteed to have 1 element in it?

Reply:
I have added it as a warning but unsure if it is necessary to error the validation on this case.

Reply:
Warning seems fine since the task could presumably still run even if they give extra containers.